internal/testhelp/fakes/fabric_gateway.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MPL-2.0
+
+package fakes
+
+import (
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	fabcore "github.com/microsoft/fabric-sdk-go/fabric/core"
+	fabfake "github.com/microsoft/fabric-sdk-go/fabric/fake"
+
+	gw "github.com/microsoft/terraform-provider-fabric/internal/services/gateway"
+	"github.com/microsoft/terraform-provider-fabric/internal/testhelp"
+)
+
+// operationsGateway implements SimpleIDOperations.
+type operationsGateway struct{}
+
+func (o *operationsGateway) Create(data fabcore.CreateGatewayRequestClassification) fabcore.GatewayClassification {
+	switch gateway := data.(type) {
```

Comment:
```suggestion
	switch data := data.(type) {
```

the you do not have to aliasing gateway package